File: newcode/剑指Offer/1.二维数组中的查找/Solution.py
# -*- coding:utf-8 -*-
class Solution:
    # array 二维列表
    def Find(self, target, array):
        rows = len(array)
        columns = len(array[0])
        if rows == 0 or columns == 0:
            return False
        row = rows -1
        column = 0
        while( row >= 0 and column <= columns -1):
            if array[row][column] < target:
                column +=1
            elif array[row][column] > target:
                row -=1
            else :
                return True
        return False

if __name__ == '__main__':
    arrays =[[1,2,8,9],[2,4,9,12],[4,7,10,13],[6,8,11,15]]
    print(Solution().Find(11,arrays))

File: newcode/剑指Offer/1.二维数组中的查找/Solution.java
public class Solution {
    public boolean Find(int target, int [][] array) {
        int rows = array.length;
        if (rows == 0){
            return false;
        }
        int columns = array[0].length;
        if(columns == 0){
            return false;
        }
        
        int row = rows-1;
        int column = 0;
        
        while(row >= 0 && column <= columns -1){
            if(array[row][column] < target){
                column ++;
            } else if( array[row][column] > target){
                row --;
            } else{
                return true;
            }
        }
        return false;
    }
}